encoder-brave/src/main/java/zipkin2/reporter/otel/brave/SpanTranslator.java

```diff
@@ -0,0 +1,212 @@
+  static final Map<String, String> RENAMED_ATTRIBUTES;
+
+  static {
+    RENAMED_ATTRIBUTES = new LinkedHashMap<>();
+    // "http.host" is not defined in HttpTags, but is a well-known tag.
+    RENAMED_ATTRIBUTES.put("http.host", ServerAttributes.SERVER_ADDRESS.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.METHOD.key(), HttpAttributes.HTTP_REQUEST_METHOD.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.PATH.key(), UrlAttributes.URL_PATH.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.ROUTE.key(), HttpAttributes.HTTP_ROUTE.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.URL.key(), UrlAttributes.URL_FULL.getKey());
+    RENAMED_ATTRIBUTES.put(HttpTags.STATUS_CODE.key(), HttpAttributes.HTTP_RESPONSE_STATUS_CODE.getKey());
+  }
+
+  private final Consumer consumer;
+
+  SpanTranslator(Tag<Throwable> errorTag) {
+    this.consumer = new Consumer(errorTag, RENAMED_ATTRIBUTES);
+  }
+
+  TracesData translate(MutableSpan braveSpan) {
+    TracesData.Builder tracesDataBuilder = TracesData.newBuilder();
+    Builder resourceSpansBuilder = ResourceSpans.newBuilder();
+    ScopeSpans.Builder scopeSpanBuilder = ScopeSpans.newBuilder();
+    Span.Builder spanBuilder = builderForSingleSpan(braveSpan, resourceSpansBuilder);
+    scopeSpanBuilder.addSpans(spanBuilder
+        .build());
+    InstrumentationScope.Builder scopeBuilder = InstrumentationScope.newBuilder();
+    scopeBuilder.setName(BraveScope.getName());
+    scopeBuilder.setVersion(BraveScope.getVersion());
+    scopeSpanBuilder.setScope(scopeBuilder.build());
+    resourceSpansBuilder.addScopeSpans(scopeSpanBuilder
+        .build());
+    tracesDataBuilder.addResourceSpans(resourceSpansBuilder.build());
+    return tracesDataBuilder.build();
+  }
+
+  static KeyValue stringAttribute(String key, String value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setStringValue(value))
+        .build();
+  }
+
+  static KeyValue intAttribute(String key, int value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setIntValue(value))
+        .build();
+  }
+
+  private Span.Builder builderForSingleSpan(MutableSpan span, Builder resourceSpansBuilder) {
+    Span.Builder spanBuilder = Span.newBuilder()
+        .setTraceId(span.traceId() != null ? ByteString.fromHex(span.traceId()) : INVALID_TRACE_ID)
+        .setSpanId(span.id() != null ? ByteString.fromHex(span.id()) : INVALID_SPAN_ID);
+    if (span.name() != null) {
+      spanBuilder.setName(span.name());
+    }
+    if (span.parentId() != null) {
+      spanBuilder.setParentSpanId(ByteString.fromHex(span.parentId()));
+    }
+    long start = span.startTimestamp();
+    long finish = span.finishTimestamp();
+    spanBuilder.setStartTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(start));
+    spanBuilder.setEndTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(finish));
+    Kind kind = span.kind();
+    if (kind != null) {
+      switch (kind) {
```

Comment:
Nit, can consider extracting a method to convert kind, then no need for `break` lines.

Reply:
Fixed in https://github.com/openzipkin-contrib/zipkin-otel/pull/9/commits/81f3a622b42074ef966163c68fc9614c1585afef